test/t8_test_successor.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element types in parallel.
+
+  Copyright (C) 2010 The University of Texas System
+  Written by Carsten Burstedde, Lucas C. Wilcox, and Tobin Isaac
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <t8_eclass.h>
+#include <t8_schemes/t8_default_cxx.hxx>
+
+/* Check the computation of the successor recursively. Iterate through the elements
+ * via DFS. On the given maxlvl-1 the children are computeted iteratively. For
+ * each child, the successor is checked.
+ */
+static void
+t8_recursive_successor (t8_element_t * element, t8_element_t * successor,
+                        t8_element_t * child,
+                        t8_element_t * last, t8_eclass_scheme_c * ts,
+                        int level, const int maxlvl)
```

Comment:
The level parameter is redundant since we can get it through t8_element_level and should hence remove it.